x-pack/plugins/rule_registry/server/rule_data_plugin_service/resource_installer.ts

```diff
@@ -112,16 +110,15 @@ export class ResourceInstaller {
                   name: DEFAULT_ALERTS_ILM_POLICY_NAME,
                   body: DEFAULT_ALERTS_ILM_POLICY,
                 }),
+                this.createOrUpdateComponentTemplate({
```

Comment:
When framework alerts are enabled, the later `install` functions (`installIndexLevelResources` and `installNamespacedIndexTemplate`) that depend on `installCommonResources` succeeding no longer have the dependency on the ILM policy or ECS templates being installed. As a result, failures or delays in installation of these resources, when handled by the framework, could cause alerts indices to be created incorrectly.

Reply:
👍 Thanks for pointing this out! This is a first in a series of PRs to move resource installation to the alerting plugin. When we finish this phase, the feature flag will determine which plugin handles all of the installation, not just pieces of it, so there should not be a case where the framework installs the common component templates but the rule registry is installing index level and namespace level resources.